src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,122 +1,241 @@
+    }
+
+    public StudentList(int initialCapacity) {
+        CAPACITY = initialCapacity;
+        updateCapacity(elements, initialCapacity);
+    }
+
+    public StudentList(Collection<? extends Student> studentsCollection) {
+        addAll(studentsCollection);
+    }
 
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return sizeOfArray;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return sizeOfArray == 0;
     }
 
     @Override
     public boolean contains(Object o) {
-        //TODO
-        return false;
+        return indexOf(o) != -1 ? true : false;
     }
 
     @Override
     public Iterator<Student> iterator() {
-        //TODO
-        return null;
+        return new Iterator<Student>() {
+            private int initial = 0;
+
+            @Override
+            public boolean hasNext() {
+                return elements.length > initial;
+            }
+
+            @Override
+            public Student next() {
+                return elements[initial++];
+            }
+        };
     }
 
     @Override
     public Object[] toArray() {
-        //TODO
-        return null;
+        return Arrays.copyOf(elements, sizeOfArray);
     }
 
     @Override
     public <T> T[] toArray(T[] ts) {
-        //TODO
-        return null;
+        if (ts.length < size()) {
+            return (T[]) Arrays.copyOf(elements, size(), ts.getClass());
+        }
+        System.arraycopy(elements, 0, ts, 0, ts.length);
+        if (ts.length > size()) {
+            ts[size()] = null;
+        }
+        return ts;
     }
 
     @Override
     public boolean add(Student student) {
-        //TODO
-        return false;
+        checkCapacity();
+        elements[sizeOfArray++] = student;
+        return true;
+    }
+
+    private void checkCapacity() {
+        if (elements.length == CAPACITY || elements.length > CAPACITY) {
+            CAPACITY = elements.length;
+            int capacity = CAPACITY + 10;
+            updateCapacity(elements, capacity);
+        }
+    }
+
+    private void updateCapacity(Student[] students, int newSize) {
+        elements = Arrays.copyOf(students, newSize);
     }
 
     @Override
     public boolean remove(Object o) {
-        //TODO
+        int index = indexOf(o);
+        if (index != -1) {
+            System.arraycopy(elements, index + 1, elements, index, size() - index + 1);
+            sizeOfArray--;
+            return true;
+        }
         return false;
     }
 
     @Override
     public void clear() {
-        //TODO
+        for (int i = 0; i < size(); i++) {
+            elements[i] = null;
+        }
+        CAPACITY = 10;
+        sizeOfArray = 0;
     }
 
     @Override
     public Student get(int i) {
-        //TODO
-        return null;
+        checkTheExistanceOfIndex(i);
+        return elements[i];
+    }
+
+    private boolean checkTheExistanceOfIndex(int index) {
```

Comment:
This method throws an exception if index is out of bound, then it returns true.
I think we can remove this return value, which will reduce if statement where we use the result of this method.

This method can return void and just throw an exception if needed.